tests/parser/features/external_contracts/test_external_contract_calls.py

```diff
@@ -222,7 +222,640 @@ def get_array(arg1: address) -> (Bytes[30], int128, Bytes[3]):
+    return 2
+    """
+
+    c = get_contract(contract_1)
+
+    contract_2 = """
+interface Foo:
+    def foo() -> bool: view
+
+@external
+def bar(arg1: address) -> bool:
+    return Foo(arg1).foo()
+"""
+
+    c2 = get_contract(contract_2)
+    assert_tx_failed(lambda: c2.bar(c.address))
+
+
+@pytest.mark.parametrize("a", ["uint8", "uint256", "int128", "int256"])
+@pytest.mark.parametrize("b", ["uint8", "uint256", "int128", "int256"])
+def test_tuple_with_bool(get_contract, a, b):
+    contract_1 = f"""
+@external
+def foo() -> ({a}, Bytes[3], {b}):
+    return 1, b'dog', 0
+    """
+
+    c = get_contract(contract_1)
+
+    contract_2 = """
+interface Foo:
+    def foo() -> (bool, Bytes[3], bool): view
+
+@external
+def bar(arg1: address) -> (bool, Bytes[3], bool):
+    a: bool = False
+    b: Bytes[3] = b""
+    c: bool = False
+    a, b, c = Foo(arg1).foo()
+    return a, b, c
+"""
+
+    c2 = get_contract(contract_2)
+    assert c.foo() == [1, b"dog", 0]
+    assert c2.bar(c.address) == [True, b"dog", False]
+
+
+@pytest.mark.parametrize("a", ["uint8", "uint256", "int128", "int256"])
+@pytest.mark.parametrize("b", ["uint8", "uint256", "int128", "int256"])
+def test_tuple_with_bool_too_long(get_contract, assert_tx_failed, a, b):
+    contract_1 = f"""
+@external
+def foo() -> ({a}, Bytes[3], {b}):
+    return 1, b'dog', 2
+    """
+
+    c = get_contract(contract_1)
+
+    contract_2 = """
+interface Foo:
+    def foo() -> (bool, Bytes[3], bool): view
+
+@external
+def bar(arg1: address) -> (bool, Bytes[3], bool):
+    a: bool = False
+    b: Bytes[3] = b""
+    c: bool = False
+    a, b, c = Foo(arg1).foo()
+    return a, b, c
+"""
+
+    c2 = get_contract(contract_2)
+    assert c.foo() == [1, b"dog", 2]
+    assert_tx_failed(lambda: c2.bar(c.address))
+
+
+@pytest.mark.parametrize("type", ["uint8", "int128", "uint256", "int256"])
+def test_external_contract_calls_with_address(get_contract, type):
+    contract_1 = f"""
+@external
+def foo() -> {type}:
+    return 1
+    """
+
+    c = get_contract(contract_1)
+
+    contract_2 = """
+interface Foo:
+    def foo() -> address: view
+
+@external
+def bar(arg1: address) -> address:
+    return Foo(arg1).foo()
+"""
+
+    c2 = get_contract(contract_2)
+    assert c2.bar(c.address) == "0x0000000000000000000000000000000000000001"
+
+
+@pytest.mark.parametrize("type", ["uint256", "int256"])
+def test_external_contract_calls_with_address_two(get_contract, type):
+    contract_1 = f"""
+@external
+def foo() -> {type}:
+    return (2**160)-1
+    """
+
+    c = get_contract(contract_1)
+
+    contract_2 = """
+interface Foo:
+    def foo() -> address: view
+
+@external
+def bar(arg1: address) -> address:
+    return Foo(arg1).foo()
+"""
+
+    c2 = get_contract(contract_2)
+    assert c2.bar(c.address).lower() == "0xffffffffffffffffffffffffffffffffffffffff"
+
+
+@pytest.mark.parametrize("type", ["uint256", "int256"])
+def test_address_too_long(get_contract, assert_tx_failed, type):
```

Comment:
I wonder if there is some way to parametrize these tests across types?

Reply:
Do you mean something like this: `def test_{type}_too_long()`?